.github/workflows/test-changed-metadata.yml

```diff
@@ -47,6 +47,15 @@ jobs:
     steps:
       - name: "☁️ Checkout repository"
         uses: actions/checkout@v3
+      - name: "Pull allowed docker images"
+        run: |
+          ./gradlew pullAllowedDockerImages
```

Comment:
This means we unconditionally pull X docker images, even if the build job doesn't need it. We should only download the image(s) that are actually needed.

Reply:
This is fixed. Only images specified by the test we are running, will be pulled.